### PR TITLE
Size and data offset helper function for rlp lists

### PR DESCRIPTION
**Pull request by gballet**
Factor some size/offset deserialization code.

### CODE REVIEW

File: src/deserialize.zig [hunk 1]
```diff
@@ -103,19 +124,23 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T) !usize {
...
+                //     return 1 + size;
+                // } else {
+                //     const size_size = @as(usize, serialized[0] - rlpByteListLongHeader);
+                //     const size = readIntSliceBig(usize, serialized[1 .. 1 + size_size]);
+                //     out.* = serialized[1 + size_size .. 1 + size_size + size];
+                //     return 1 + size + size_size;
+                // }
+                var r = sizeAndDataOffset(serialized);
+                std.debug.print("{} {}\n", .{ r.offset, r.size });
+                out.* = serialized[r.offset .. r.offset + r.size];
```
Comment (jsign) [inline src/deserialize.zig:108]: [Cross-linking comment.](https://github.com/gballet/zig-rlp/pull/11#issuecomment-1849935568)

Comment (gballet) [reply src/deserialize.zig:108]: yup that makes sense. I'll do it in another PR though.

File: src/deserialize.zig [hunk 2]
```diff
@@ -103,19 +104,10 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T) !usize {
...
-                    out.* = serialized[1 .. 1 + size];
-                    return 1 + size;
-                } else {
-                    const size_size = @as(usize, serialized[0] - rlpByteListLongHeader);
-                    const size = readIntSliceBig(usize, serialized[1 .. 1 + size_size]);
-                    out.* = serialized[1 + size_size .. 1 + size_size + size];
-                    return 1 + size + size_size;
-                }
+                var r = sizeAndDataOffset(serialized);
+                std.debug.print("{} {}\n", .{ r.offset, r.size });
```
Comment (gballet) [inline src/deserialize.zig]:
```suggestion
```

File: src/deserialize.zig [hunk 3]
```diff
@@ -374,3 +357,27 @@ test "access list empty" {
...
+//     };
+//     const StrippedTxn = struct {
+//         access_list: []AccessListItem,
+//     };
+
+//     var buf: [128]u8 = undefined;
+//     const rlp = try std.fmt.hexToBytes(&buf, "f83af838f7940000000000000000000000000000000000001210e1a00000000000000000000000000000000000000000000000000000000000000203");
+//     var out: StrippedTxn = StrippedTxn{ .access_list = &[0]AccessListItem{} };
+//     _ = try deserialize(StrippedTxn, rlp, &out);
+// }
```
Comment (gballet) [inline src/deserialize.zig]:
```suggestion
```

**Pull request merged**